lib/bolt/inventory.rb

```diff
@@ -88,16 +92,37 @@ def groups_in(node_name)
       @groups.data_for(node_name)['groups'] || {}
     end
 
+    def transform_data(data)
+      if data
+        data['config'] = Bolt::Util.symbolize_keys(data['config']) if data['config']
+      end
+      data
+    end
+    private :transform_data
+
     # Pass a target to get_targets for a public version of this
     # Should this reconfigure configured targets?
     def update_target(target)
-      inv_conf = config_for(target.host)
-      unless inv_conf
-        @logger.debug("Did not find #{target.host} in inventory")
-        inv_conf = {}
+      data = @groups.data_for(target.host)
+      processed = transform_data(data) || {}
+
+      unless processed['config']
+        @logger.debug("Did not find config for #{target.host} in inventory")
+        processed['config'] = {}
       end
 
-      conf = Bolt::Util.deep_merge(@config.transport_conf, inv_conf)
+      unless processed['vars']
+        @logger.debug("Did not find any variables for #{target.host} in inventory")
+        processed['vars'] = {}
+      end
+
+      tvars = @target_vars[target.host] || {}
+      merged = Bolt::Util.deep_merge(tvars, processed['vars'])
```

Comment:
Won't this overwrite vars a user has set on the target from puppet code next time they call run_*?